bindings/rust/wasmedge-sdk/examples/host_func_with_custom_error.rs

```diff
@@ -0,0 +1,125 @@
+//! In this example, we'll present how to define a custom error type and use it in our host function.
+//! To run the example, just follow the commands below
+//!
+//! ```bash
+//! cd wasmedge-sdk/
+//! cargo run --example host_func_with_custom_error
+//! ```
+//!
+//! After running the example, you should see the following output:
+//!
+//! ```bash
+//! User error: OutOfUpperBound
+//! ```
+//! which means that the one or two input numbers of i32 type of the host function are out of upper bound (here is 100).
+#![feature(never_type)]
+
+use thiserror::Error;
+use wasmedge_sdk::{
+    config::ConfigBuilder,
+    error::{HostFuncError, WasmEdgeError},
+    host_function, params,
+    types::ExternRef,
+    Caller, ImportObjectBuilder, ValType, Vm, WasmVal, WasmValue,
+};
+
+// Define custom error type
+#[derive(Error, Clone, Debug, PartialEq, Eq)]
+enum MyError {
+    #[error("The number of arguments should be 3")]
+    InvalidArguments,
+    #[error("The type of the first argument should be i32")]
+    InvalidFirstArgumentType,
+    #[error("The type of the second argument should be i32")]
+    InvalidSecondArgumentType,
+    #[error("The value of argument is out of upper bound.")]
+    OutOfUpperBound,
+}
+impl From<MyError> for u32 {
```

Comment:
Not sure if we have to do this by hard-coding.

May the `IntEnum` help?
https://docs.rs/int-enum/latest/int_enum/trait.IntEnum.html

Reply:
This is just to demonstrate a possible way for developers to define a custom error type and use it. It finally depends on the developers how to do it.